packages/web3-utils/src/validation.ts

```diff
@@ -109,46 +110,36 @@ export const isTopicInBloom = isTopicInBloomValidator;
  */
 export const compareBlockNumbers = (blockA: Numbers, blockB: Numbers) => {
 	// string validation
-	if (
-		typeof blockA === 'string' &&
-		!(
-			blockA === 'genesis' ||
-			blockA === 'earliest' ||
-			blockA === 'pending' ||
-			blockA === 'latest'
-		)
-	)
-		throw new InvalidBlockError(blockA);
-	if (
-		typeof blockB === 'string' &&
-		!(
-			blockB === 'genesis' ||
-			blockB === 'earliest' ||
-			blockB === 'pending' ||
-			blockB === 'latest'
-		)
-	)
-		throw new InvalidBlockError(blockB);
+	if (blockA === 'genesis' || blockB === 'genesis')
+		throw new InvalidBlockError('Genesis tag not supported'); // for more specific error message
+	if (typeof blockA === 'string' && !isBlockTag(blockA)) throw new InvalidBlockError(blockA);
+	if (typeof blockB === 'string' && !isBlockTag(blockB)) throw new InvalidBlockError(blockB);
+
+	// Increasing order:  earliest, safe, (finalized ~ latest), pending
```

Comment:
It seems this is the order according to https://www.alchemy.com/overviews/ethereum-commitment-levels and https://hackmd.io/@prysmaticlabs/finality and so, the conditions below need to be updated (I think the conditions for `finalized`  needs to be modified.)
```suggestion
	// Increasing order:  earliest, finalized, safe (justified), latest, pending
```

Reply:
modified